iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi_multi_process.hpp

```diff
@@ -41,33 +41,33 @@ using namespace iox::units::duration_literals;
 class RouDiMultiProcess
 {
   public:
-
     // indicate whether the message queue thread will start directly or deferred
     // this is important for derived classes which may need to initialize their members before the thread starts
-    enum class MQThreadStart {
-      IMMEDIATE,
-      DEFER_START
+    enum class MQThreadStart
+    {
+        IMMEDIATE,
+        DEFER_START
     };
-   
+
 
     RouDiMultiProcess& operator=(const RouDiMultiProcess& other) = delete;
     RouDiMultiProcess(const RouDiMultiProcess& other) = delete;
 
-    RouDiMultiProcess(RouDiMemoryInterface& roudiMemoryInteface,
-                      PortManager& portManager,
-                      const MonitoringMode f_monitoringMode = MonitoringMode::ON,
-                      const bool f_killProcessesInDestructor = true,
-                      const MQThreadStart mqThreadStart = MQThreadStart::IMMEDIATE);
+    RouDiMultiProcess(
+        RouDiMemoryInterface& roudiMemoryInteface,
+        PortManager& portManager,
+        const MonitoringMode f_monitoringMode = MonitoringMode::ON,
+        const bool f_killProcessesInDestructor = true,
+        const MQThreadStart mqThreadStart = MQThreadStart::IMMEDIATE,
+        const version::CompatibilityCheckLevel compatibilityCheckLevel = version::CompatibilityCheckLevel::PATCH);
```

Comment:
question to all, what compatibility level do we want to have by default? Should 0.18.1 be incompatible to 0.18.2?

Reply:
I think the external API should be compatible among all 0.18.x versions. The x should only increase for non API breaking bugfixes.

Reply:
This is more about ABI than API. We want to prevent/detect subtle breakages due to binary incompatible changes